perception/sub8_perception/nodes/orange_rectangle_finder.py

```diff
@@ -279,30 +230,30 @@ def _update_kf(self, (x, y, z, dy, dx)):
                   rospy.loginfo("Marker Found")
               self.found = True
 
-    def _get_pose_3D(self, p):
-        i_points = np.array(p,dtype=np.float)
-        # Use camera intrinsics and knowledge of marker's real demensions to get a pose estimate in camera frame
-        _, rvec, tvec =  cv2.solvePnP(PathMarkerFinder.PATH_MARKER, i_points, self.cam.intrinsicMatrix(), np.zeros((5,1)))
+    def _get_pose_3D(self, corners):
+        tvec, rvec = self.rect_model.get_pose_3D(corners, cam=self.cam, rectified=True)
         if tvec[2][0] < 0.3: # Sanity check on position estimate
             rospy.logwarn("Marker too close, must be wrong...")
             return False
+        rmat, _ = cv2.Rodrigues(rvec)
+        vec = rmat.dot(np.array([1, 0 , 0]))
```

Comment:
You can also take the first column of the rmat to get the forward vector (which is what this dot product does)